src/platform/packages/shared/kbn-telemetry/src/init_telemetry.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+import { loadConfiguration } from '@kbn/apm-config-loader';
+import { LateBindingSpanProcessor } from '@kbn/tracing';
+import { context, trace } from '@opentelemetry/api';
+import { AsyncLocalStorageContextManager } from '@opentelemetry/context-async-hooks';
+import { resourceFromAttributes } from '@opentelemetry/resources';
+import {
+  NodeTracerProvider,
+  ParentBasedSampler,
+  TraceIdRatioBasedSampler,
+} from '@opentelemetry/sdk-trace-node';
+import { ATTR_SERVICE_NAME, ATTR_SERVICE_VERSION } from '@opentelemetry/semantic-conventions';
+
+/**
+ *
+ * Initializes OpenTelemetry (currently only tracing)
+ *
+ * @param argv                Process arguments
+ * @param rootDir             Root dir of Kibana repo
+ * @param isDistributable     Whether this is a distributable build
+ * @param serviceName         The service name used in resource attributes
+ * @returns                   A function that can be called on shutdown and allows exporters to flush their queue.
+ */
+export const initTelemetry = (
```

Comment:
@dgieselaar Is there a problem you see when we have both OTel SDK and APM Node.js agent active at the same time? Should we set `active` flag to false when we enable OTel SDK?

Reply:
All auto-instrumentations for `@opentelemetry` should be disabled. Only spans in the context of Inference calls (chatComplete) should be recorded and exported, which are relatively low in volume. [The elasticsearch-js client always creates spans via Transport](https://github.com/elastic/elastic-transport-js/blob/bb2e9c061f68b84313dc54c1564d43e04d752f94/src/Transport.ts#L694), so those will be recorded as well - kind of accidentally, I'm not sure that's how it should work, but in this case also convenient. So they should be able to run in parallel without any problems, but if we migrate more stuff to OTel I think we should consider one or the other.

Reply:
Gotcha, thanks for the context and agree we need to move towards the OTel SDK soonish.